### ISSUE TITLE
SSL error message when running tzupdate with python 3.8

### ISSUE THREAD

**Issue by smac89** [cdown/tzupdate#35]
Here is the output of running `sudo tzupdate`:

```
Process Process-4:
Process Process-2:
Traceback (most recent call last):
  File "/usr/lib/python3.8/urllib/request.py", line 1319, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/usr/lib/python3.8/http/client.py", line 1230, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.8/http/client.py", line 1276, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.8/http/client.py", line 1225, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.8/http/client.py", line 1004, in _send_output
    self.send(msg)
  File "/usr/lib/python3.8/http/client.py", line 944, in send
    self.connect()
  File "/usr/lib/python3.8/http/client.py", line 1399, in connect
    self.sock = self._context.wrap_socket(self.sock,
  File "/usr/lib/python3.8/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/lib/python3.8/ssl.py", line 1040, in _create
    self.do_handshake()
  File "/usr/lib/python3.8/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1108)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.8/site-packages/tzupdate.py", line 84, in get_timezone_for_ip
    api_response_obj = urlopen(api_url)  # nosec
  File "/usr/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/usr/lib/python3.8/urllib/request.py", line 542, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 1362, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "/usr/lib/python3.8/urllib/request.py", line 1322, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1108)>
Traceback (most recent call last):
  File "/usr/lib/python3.8/urllib/request.py", line 1319, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/usr/lib/python3.8/http/client.py", line 1230, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.8/http/client.py", line 1276, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.8/http/client.py", line 1225, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.8/http/client.py", line 1004, in _send_output
    self.send(msg)
  File "/usr/lib/python3.8/http/client.py", line 944, in send
    self.connect()
  File "/usr/lib/python3.8/http/client.py", line 1399, in connect
    self.sock = self._context.wrap_socket(self.sock,
  File "/usr/lib/python3.8/ssl.py", line 500, in wrap_socket
    return self.sslsocket_class._create(
  File "/usr/lib/python3.8/ssl.py", line 1040, in _create
    self.do_handshake()
  File "/usr/lib/python3.8/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1108)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.8/site-packages/tzupdate.py", line 84, in get_timezone_for_ip
    api_response_obj = urlopen(api_url)  # nosec
  File "/usr/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/usr/lib/python3.8/urllib/request.py", line 542, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 1362, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "/usr/lib/python3.8/urllib/request.py", line 1322, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1108)>
Set system timezone to America/Regina.
```

It still works, but the error message is disconcerting.

Installed from [AUR](https://aur.archlinux.org/packages/tzupdate/)

**Comment by cdown** [comment]
The only two services we talk to over SSL are https://ipapi.co and https://freegeoip.app/, and both are serving valid certificates now. Does this still happen?

**Comment by smac89** [comment]
I get the following error when attempting to visit `freegeoip.app` using brave browser:
![image](https://user-images.githubusercontent.com/8305511/73459129-e9a11180-433b-11ea-941c-1becf74fc455.png)
I also get a similar issue when attempting to visit `ipapi.co`.

Strangely enough, both of those work in firefox, but attempting to even curl any results in a similar SSL error.
At this point I'm not sure what the scope of this issue is, and if it is limited to just my machine.

**Comment by cdown** [comment]
What is the cert chain presented? Are you sure your connection isn't being fiddled with (captive portal, etc, etc?)

**Comment by smac89** [comment]
I finally figured it out. I tried pinging ipapi.co and saw that the ip address was being resolved to 127.0.0.1 which I found to be strange.
After scratching my head for a while, I checked my `/etc/hosts` file and saw that `ipapi.co` was being blocked and redirected to `0.0.0.0`. I am running [`hblock`](https://aur.archlinux.org/packages/hblock/) and for some reason it had blocked both of the urls and the only url not being blocked was http://worldtimeapi.org. 

I'm still not sure why it worked with Firefox and not with Brave/Chrome.
Anyways, the error message was quite confusing because I began to suspect something was wrong with my browser when all along it was just a dumb hosts configuration.

I suggest catching this error in the script and logging an appropriate error message rather than letting the exception bubble up to the user. I can create a PR for this if you want.

**Comment by cdown** [comment]
Great! I'm glad you got it sorted.

> I suggest catching this error in the script and logging an appropriate error message rather than letting the exception bubble up to the user.

An exception is already the right way to handle such an unexpected situation. We can't really handle every single case in which the operating environment might be pathological :-)
